force-app/main/instrumentation/classes/SfdoInstrumentationEnum.cls

```diff
@@ -0,0 +1,130 @@
+/*
+    Copyright (c) 2021 Salesforce.org
+    All rights reserved.
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2021
+* @group Product Instrumentation
+* @description Enums of Features, Components and Actions to use with product instrumentation. These ensure consistency
+* of values in the Splunk logs.
+*
+*/
+public with sharing class SfdoInstrumentationEnum {
+
+    /**
+     * @description High Level product feature for the instrumentation entries. New Features can be added to this list when required.
+     * Always be sure to avoid duplicate or ambiguous feature names. Features defined here will be queryable in SFDC's internal system.
+     */
+    public enum Feature {
+        TDTM,
+        ErrorHandling,
+        Addresses,
+        GetStarted,
+        AccountModelConversion,
+        Settings,
+        EnhancedRecurringDonations,
+        LegacyRecurringDonations,
+        GiftEntry,
+        DataImportEngine,
+        EngagementPlans,
+        Households,
+        Payments,
+        Allocations,
+        ContactRoles,
+        Donations,
+        Relationships,
+        Affiliations,
+        SoftCreditAutomation,
+        MatchingGifts,
+        PartialSoftCredits,
+        CustomizableRollups,
+        LegacyRollups,
+        PaymentServices,
+        LeadConversion,
+        RecordMerge
+    }
```

Comment:
@force2b Where would things like CallableAPI and Elevate setup/usage fall into this list?  Maybe no need to think about it at this point since it can be easily modified.

Reply:
We can add features to this list at any point. The CallableApi stuff I think we'd just put under the related feature, unless we wanted to explicitly instrument usage of the Callable Api's. In that case, then maybe we'd add a new Feature for it or maybe just one for Api in general. Elevate would probably get added as well - especially if we wanted to capture data on outbound api calls for authentication/charging/etc.